### PR TITLE
ci(gitlab): update to autodevops 20.7.14

### PR DESCRIPTION
**Pull request by douglasduteil**

### CODE REVIEW

File: .gitlab-ci.yml [hunk 1]
```diff
@@ -15,122 +12,82 @@ include:
...
+    GIT_STRATEGY: none
+  script:
+    - yarn config set cache-folder ${CI_PROJECT_DIR}/.cache/yarn
+    - yarn global add degit
+    - yarn exec degit "SocialGouv/kosko-charts/templates/autodevops#v5.2.0-alpha.5" .k8s
+    - yarn --cwd .k8s --production --frozen-lockfile --prefer-offline --link-duplicates
+    - yarn --silent --cwd .k8s dropdb > ${CI_PROJECT_DIR}/manifest.yaml
+    - kapp -y deploy -a ${CI_PROJECT_NAME} --namespace ${KUBE_NAMESPACE} -f ${CI_PROJECT_DIR}/manifest.yaml
+    - kapp -y delete -a ${CI_PROJECT_NAME} --namespace ${KUBE_NAMESPACE}
+    - kubectl delete namespace "${KUBE_NAMESPACE}" || true
```
Comment (revolunet) [inline .gitlab-ci.yml:77]: on peut extraire ca gitlab-ci-ym ?

File: .k8s/__tests__/generate-feature.ts [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+      CI_COMMIT_REF_NAME: "jean-mich/Jeune-homme-à--la-roses-1.2.3",
+      CI_COMMIT_REF_SLUG: "jean-mich-jeune-homme-à--la-roses-1-2-3",
+      CI_ENVIRONMENT_NAME: "jean-mich/Jeune-homme-a-la-roses-1.2.3",
+      CI_ENVIRONMENT_SLUG: "jean-mich-jeune-123456",
+      CI_ENVIRONMENT_URL: `https://jean-mich-jeune-123456-cdtn-admin.dev2.fabrique.social.gouv.fr`,
+      KUBE_NAMESPACE: `cdtn-admin-85-jean-mich-jeune-123456`,
+      //
+      RANCHER_PROJECT_ID: "c-bar:p-foo",
+    })
+  ).toMatchSnapshot();
```
Comment (revolunet) [inline .k8s/__tests__/generate-feature.ts:22]: isnt generate-dev a default feature env ? 🤔

File: .k8s/__tests__/generate-prod.ts [hunk 3]
```diff
@@ -8,6 +8,7 @@ test("kosko generate --prod", async () => {
   expect(
     await getEnvManifests("prod", "", {
       ...project("cdtn-admin").prod,
+      IP_ALLOWLIST: "123.456.456.789,42.0.0.0/8",
```
Comment (revolunet) [inline .k8s/__tests__/generate-prod.ts:11]: ca typiquement (comme le RANCHER_PROJECT_ID) ce serait bien de ne pas devoir le mettre en dur :/ ca oblige d'avoir les valeurs sync avec gitlab

File: .k8s/components/hasura.ts [hunk 4]
```diff
@@ -0,0 +1,44 @@
...
+  env,
+});
+
+if (env.env === "dev") {
+  // HACK(douglasduteil): provide one db per env
+  // The CI_ENVIRONMENT_SLUG is the most useful for this
+  const pgSecretRefName = `azure-pg-user-${GITLAB_LIKE_ENVIRONMENT_SLUG.replace(
+    /-/g,
+    ""
+  )}`;
```
Comment (revolunet) [inline .k8s/components/hasura.ts:27]: should be fixed at kosko level

File: .k8s/components/jobs/restore/container.ts [hunk 5]
```diff
@@ -1,31 +1,27 @@
...
   env: [
     new EnvVar({
       name: "SOURCE_CONTAINER",
-      value: process.env.SOURCE_CONTAINER,
+      value: "cdtn",
     }),
     new EnvVar({
       name: "DESTINATION_CONTAINER",
-      value: process.env.DESTINATION_CONTAINER,
+      value: "cdtn-dev",
```
Comment (revolunet) [inline .k8s/components/jobs/restore/container.ts:13]: we had a case where the preprod container was copied to production when "ready"

File: .k8s/components/jobs/restore/db.ts [hunk 6]
```diff
@@ -1,46 +1,71 @@
...
+const pgUserEnvVar = initContainer.env?.find((e) => e.name === "PGUSER");
+const { host } = getDefaultPgParams();
+ok(pgUserEnvVar, "Missing PGUSER variable");
+pgUserEnvVar.value = `${pgParams.user}@${host}`;
+
+const pgPasswordEnvVar = initContainer.env?.find(
+  (e) => e.name === "PGPASSWORD"
+);
+ok(pgPasswordEnvVar, "Missing PGPASSWORD variable");
+pgPasswordEnvVar.value = pgParams.password;
```
Comment (revolunet) [inline .k8s/components/jobs/restore/db.ts:69]: the initContainer should already have the correct `envFrom.secretRef` no ?

File: .k8s/components/pg.ts [hunk 7]
```diff
@@ -0,0 +1,35 @@
...
+
+  const envParams = gitlab(process.env);
+  // add gitlab annotations
+  updateMetadata(secret, {
+    annotations: envParams.annotations ?? {},
+    labels: envParams.labels ?? {},
+    namespace: envParams.namespace,
+  });
+  return [secret];
+};
```
Comment (revolunet) [inline .k8s/components/pg.ts:35]: still an open question : how to extract that to kosko-charts

File: .k8s/components/www.ts [hunk 8]
```diff
@@ -0,0 +1,51 @@
...
+        },
+      ],
+      resources: {
+        limits: {
+          cpu: "1000m",
+          memory: "560Mi",
+        },
+        requests: {
+          cpu: "5m",
+          memory: "560Mi",
```
Comment (revolunet) [inline .k8s/components/www.ts:30]:
```suggestion
          memory: "256Mi",
```

File: .k8s/utils/GITLAB_LIKE_ENVIRONMENT_SLUG.ts [hunk 9]
```diff
@@ -0,0 +1,57 @@
...
+// based on name (which must be unique). To compensate, we add a predictable
+// 6-byte suffix in those circumstances. This is not *guaranteed* uniqueness,
+// but the chance of collisions is vanishingly small
+function suffix(name: string): string {
+  const hex = Buffer.from(
+    createHash("sha256").update(name).digest("hex")
+  ).toString();
+
+  return parseInt(hex, 16).toString(36).slice(0, 6);
+}
```
Comment (revolunet) [inline .k8s/utils/GITLAB_LIKE_ENVIRONMENT_SLUG.ts:57]: should go into kosko-charts

**Comment by revolunet** [comment]
La PR A-TO-MI-QUE :)

il faut qu'on règle le pb de DB par branche je pense

Et ce que ce serait pas l'occasion d'utiliser sur la version async de kosko ?

File: .gitlab-ci.yml (hunk 1 above)
Comment (douglasduteil) [reply .gitlab-ci.yml:77]:
```suggestion
```

File: .k8s/__tests__/generate-feature.ts (hunk 2 above)
Comment (douglasduteil) [reply .k8s/__tests__/generate-feature.ts:22]: Yes somehow

xxxxxx => feature
master => dev
tag => preprod

https://socialgouv.github.io/support/#/developpement

Comment (douglasduteil) [reply .k8s/__tests__/generate-feature.ts:22]: We might remove the test case in the future
For now it helps testing branches names over than "master"

File: .k8s/components/hasura.ts (hunk 4 above)
Comment (douglasduteil) [reply .k8s/components/hasura.ts:27]: Yep

File: .k8s/components/jobs/restore/container.ts (hunk 5 above)
Comment (douglasduteil) [reply .k8s/components/jobs/restore/container.ts:13]: Wat ? 
@lionelB are we still doing this preprod to prod thing ?

File: .k8s/components/jobs/restore/db.ts (hunk 6 above)
Comment (douglasduteil) [reply .k8s/components/jobs/restore/db.ts:69]: This is part of the "one db per branch" hack as the password here is the `azure-pg-user-<branch_name>` password

File: .k8s/components/jobs/restore/container.ts (hunk 5 above)
Comment (revolunet) [reply .k8s/components/jobs/restore/container.ts:13]: from what i know restore container is used at two steps :

 - feature-branch : restore `dev/cdtn` to `dev/cdtn-dev`
 - cdtn data update : restore `dev/cdtn` to `prod/cdtn`

File: .k8s/__tests__/generate-feature.ts (hunk 2 above)
Comment (revolunet) [reply .k8s/__tests__/generate-feature.ts:22]: ok, i meant master doesnt provide additionnal feature than dev so  guess its to have a full test with the slugified branches nams

File: .k8s/components/jobs/restore/db.ts (hunk 6 above)
Comment (revolunet) [reply .k8s/components/jobs/restore/db.ts:69]: ok so i guess this should disappear soon :)

File: .k8s/components/jobs/restore/container.ts (hunk 5 above)
Comment (lionelB) [reply .k8s/components/jobs/restore/container.ts:13]: je confirme

**Pull request merged**